pkg/controller/certificates/certificate_controller.go

```diff
@@ -143,7 +143,8 @@ func (cc *CertificateController) processNextWorkItem() bool {
 	}
 	defer cc.queue.Done(cKey)
 
-	if err := cc.syncFunc(cKey.(string)); err != nil {
+	err := controller.RunSyncAndRecord(fmt.Sprintf("%s_%s", "certificate", cc.name), cKey.(string), cc.syncFunc)
```

Comment:
Earlier we generated fmt.Sprintf("certificate-%s", cc.name).
I think it would be good to remain consistent, also if certificate is constant it seems like it should be part of the format string.
I *believe* cc.name is fairly limited in terms of values (I found ~5 values). A small number is fine but it will cause problems if that expands dramatically. Might be good to add a comment to the cc.name field to try to limit how many values are put on it.